### PR TITLE
feat: add prepareWithdrawalWorkflowWithSigner 

### PR DESCRIPTION
**Pull request by yundi-fu**
# Summary
- add `prepareWithdrawalWorkflowWithSigner `
- deprecate `prepareWithdrawalWorkflow `

# Why the changes

# Things worth calling out

### CODE REVIEW

File: src/workflows/withdrawal/completeERC20Withdrawal.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ import {
 } from '../registration';
 import { getEncodeAssetInfo } from './getEncodeAssetInfo';
 
-export async function completeERC20WithdrawalWorfklow(
+export async function completeERC20WithdrawalWorkflow(
```
Comment (yundi-fu) [inline src/workflows/withdrawal/completeERC20Withdrawal.ts:17]: fix typo

File: src/workflows/workflows.ts [hunk 2]
```diff
@@ -228,6 +229,14 @@ export class Workflows {
     );
   }
 
+  public prepareWithdrawalWithSigner(params: PrepareWithdrawalRequest) {
```
Comment (carmen0208) [inline src/workflows/workflows.ts:232]:
```suggestion
  public prepareWithdrawalWithSigner(params: WalletConnect) {
```

Comment (carmen0208) [inline src/workflows/workflows.ts:232]: can we have a request: {token, quantity} as a request, and {l1signer, l2Signer} as wallet.

```suggestion
  public prepareWithdrawalWithSigner(wallet: PrepareWithdrawalRequest, request: PrepareWithdrawal) 
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,11 +11,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added `createOrderWithSigner` function to enable create order with l2signer
 - Added `cancelOrderWithSigner` function to enable cancel order with l2signer
 - Added `WalletConnection` type
+- Added `prepareWithdrawalWorkflowWithSigner` type
```
Comment (carmen0208) [inline CHANGELOG.md:14]: - Added `prepareWithdrawalWithSigner` function to enable prepare withdraw with l2Signer

File: src/types/withdrawal.ts [hunk 4]
```diff
@@ -1,4 +1,6 @@
 import { TokenType } from './token';
+import { WalletConnection } from './index';
+import { WithdrawalsApi } from '../api';
```
Comment (carmen0208) [inline src/types/withdrawal.ts:3]: is that an unnecessary import?

File: src/workflows/workflows.ts [hunk 5]
```diff
@@ -31,7 +31,7 @@ import {
   completeERC20WithdrawalWorfklow,
   completeERC721WithdrawalWorkflow,
   completeEthWithdrawalWorkflow,
-  prepareWithdrawalWorkflow,
+  prepareWithdrawalWorkflow, prepareWithdrawalWorkflowWithSigner,
```
Comment (carmen0208) [inline src/workflows/workflows.ts:34]:
```suggestion
  prepareWithdrawalWorkflow,
  prepareWithdrawalWorkflowWithSigner,
```

File: src/workflows/workflows.ts [hunk 6]
```diff
@@ -57,7 +57,7 @@ import {
   TokenWithdrawal,
   StarkWallet,
   L1Signer,
-  L2Signer,
+  L2Signer, PrepareWithdrawalRequest,
```
Comment (carmen0208) [inline src/workflows/workflows.ts:60]:
```suggestion
  L2Signer,
  PrepareWithdrawalRequest,
```

File: src/workflows/workflows.ts [hunk 7]
```diff
@@ -228,6 +229,15 @@ export class Workflows {
     );
   }
 
+  public prepareWithdrawalWithSigner(walletConnection:   WalletConnection,request: PrepareWithdrawalRequest) {
```
Comment (carmen0208) [inline src/workflows/workflows.ts:232]:
```suggestion
  public prepareWithdrawalWithSigner(walletConnection: WalletConnection, request: PrepareWithdrawalRequest) {
```

File: src/workflows/workflows.ts [hunk 8]
```diff
@@ -228,6 +229,15 @@ export class Workflows {
...
+    return prepareWithdrawalWorkflowWithSigner( {
+      ...walletConnection,
+      ...request,
+      withdrawalsApi: this.withdrawalsApi,
+    },
+    )
```
Comment (carmen0208) [inline src/workflows/workflows.ts:238]:
```suggestion
    return prepareWithdrawalWorkflowWithSigner({
      ...walletConnection,
      ...request,
      withdrawalsApi: this.withdrawalsApi,
    })
```

**Review by carmen0208 — APPROVED** [review]

**Review by ABashkirtseva — APPROVED** [review]

**Review by zatlite — APPROVED** [review]

**Pull request merged**